### ISSUE TITLE
@code_native shows incorrect function label in `movabsq` in Julia >= 1.4

### ISSUE THREAD

**Issue by fingolfin** [JuliaLang/julia#39661]
Consider the following session (run in Julia 1.5.4, but 1.4.2 and 1.6.0-rc1 give me similar results):
```julia
julia> fact(n) = n <= 1 ? 1 : n * fact(n-1)
fact (generic function with 1 method)

julia> @code_native fact(10)
	.section	__TEXT,__text,regular,pure_instructions
; ┌ @ REPL[1]:1 within `fact'
; │┌ @ REPL[1]:1 within `<='
	pushq	%rbx
; │└
; │┌ @ int.jl:441 within `<='
	cmpq	$1, %rdi
; │└
	jle	L31
	movq	%rdi, %rbx
; │ @ REPL[1]:1 within `fact'
; │┌ @ int.jl:85 within `-'
	decq	%rdi
; │└
	movabsq	$"<=;", %rax
	callq	*%rax
; │┌ @ int.jl:87 within `*'
	imulq	%rbx, %rax
; │└
	popq	%rbx
	retq
L31:
	movl	$1, %eax
	popq	%rbx
	retq
	nopw	%cs:(%rax,%rax)
; └
```

That `movabsq` should be `movabsq $fact, %rax`. Indeed, that's what I get in Julia 1.3.1 (and older):
```
julia> fact(n) = n <= 1 ? 1 : n * fact(n-1)
fact (generic function with 1 method)

julia> @code_native fact(10)
	.section	__TEXT,__text,regular,pure_instructions
; ┌ @ REPL[1]:1 within `fact'
	pushq	%rbx
	movq	%rdi, %rbx
; │┌ @ int.jl:424 within `<='
	cmpq	$1, %rbx
; │└
	jle	L32
; │┌ @ int.jl:52 within `-'
	leaq	-1(%rbx), %rdi
; │└
	movabsq	$fact, %rax
	callq	*%rax
; │┌ @ int.jl:54 within `*'
	imulq	%rbx, %rax
; │└
	popq	%rbx
	retq
L32:
	movl	$1, %eax
	popq	%rbx
	retq
	nopw	(%rax,%rax)
; └
```

**Comment by StefanKarpinski** [comment]
Any ideas what might have changed here. @vchuravy, perhaps you might know?

**Comment by vtjnash** [comment]
Symbolicating disassembly is actually very hard, since the machine code doesn't have that information. I've got a branch to fix all of this, but it needed LLVM 11 (which we recently made required) to have some of my patches. I'll polish that up and make a PR.

**Comment by vtjnash** [comment]
Oh, actually I need LLVM 12, which was just released and we don't yet fully support (https://github.com/JuliaLang/julia/pull/40263)

**Comment by vchuravy** [comment]
> Oh, actually I need LLVM 12, which was just released and we don't yet fully support (#40263)

Moving to LLVM 12 is currently blocked on Yggdrasil binaries, which I hadn't had time to look into yet https://github.com/JuliaPackaging/Yggdrasil/pull/2906
